src/worker/hwmonitor.rs

```diff
@@ -0,0 +1,50 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::io;
+use systemstat::{CPULoad, DelayedMeasurement, Memory, NetworkStats, Platform, System};
+
+#[derive(Deserialize, Serialize, Debug, Default, Clone)]
+pub struct WorkerHwState {
+    pub worker_cpu_usage: Option<CPULoad>,
+    pub worker_memory_usage: Option<Memory>,
+    pub worker_network_usage: Option<HashMap<String, Option<NetworkStats>>>,
+}
+
+pub fn sample_cpu() -> io::Result<DelayedMeasurement<CPULoad>> {
+    let sys = System::new();
+    sys.cpu_load_aggregate()
+}
+
+impl WorkerHwState {
+    pub fn initialize_empty() -> Self {
```

Comment:
There is not need for this method since you already derive `Default`, so you have access to `WorkerHwState::default()`.